src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/ElasticsearchCaseSearchOperation.java

```diff
@@ -110,12 +113,35 @@ private CaseSearchResult toCaseDetailsSearchResult(MultiSearchResult multiSearch
                 throw new BadSearchRequest(errMsg);
             }
             if (response.searchResult != null) {
+                buildCaseFieldsAggregations(response, caseFieldsAggregations, crossCaseTypeSearchRequest);
                 caseDetails.addAll(searchResultToCaseList(response.searchResult));
                 totalHits += response.searchResult.getTotal();
             }
         }
 
-        return new CaseSearchResult(totalHits, caseDetails);
+        return new CaseSearchResult(caseFieldsAggregations, totalHits, caseDetails);
+    }
+
+    private void buildCaseFieldsAggregations(
+        final MultiSearchResult.MultiSearchResponse response,
+        final List<CaseTypesResults> caseFieldsAggregations,
+        final CrossCaseTypeSearchRequest crossCaseTypeSearchRequest) {
+
+        if (response.searchResult.getJsonObject().getAsJsonObject("hits").get("hits").getAsJsonArray().size() != 0) {
+
+            final String indexName = response.searchResult.getJsonObject().getAsJsonObject("hits").get("hits")
+                .getAsJsonArray().get(0).getAsJsonObject().get("_index").toString();
+
+            caseFieldsAggregations.add(new CaseTypesResults(getNameFromIndex(indexName,
+                crossCaseTypeSearchRequest.getCaseTypeIds()),
+                response.searchResult.getTotal())
+            );
+        }
+    }
+
+    private String getNameFromIndex(final String index, List<String> caseTypeIds) {
+        final String caseTypeIdFromIndex = index.substring(0, index.indexOf("_cases")).replaceAll("\"", "");
```

Comment:
Maybe we should use the `search.cases.index.name.format` property to determine what to remove instead, since technically it is configurable.

Reply:
DONE